risc0/zkos/v1compat/src/main.rs

```diff
@@ -198,11 +199,17 @@ mod zkvm {
     }
 
     #[no_mangle]
-    unsafe extern "C" fn ecall_software(fd: u32, mut buf: *const u8, mut len: u32) -> ! {
+    unsafe extern "C" fn ecall_software(fd: u32, mut buf: *const u8, mut len: u32) {
         // use no_std_strings::{str256, str_format};
         // let msg = str_format!(str256, "ecall_software_slow({fd:#010x}, {buf:?}, {len})");
         // print(msg.to_str());
 
+        if read_syscall_ptr(fd) != b"risc0_zkvm_platform::syscall::nr::SYS_READ" {
```

Comment:
This is a good defensive move, but I wonder if this will cause perf regressions now that every software ecall has to go thru the string compare.

Reply:
thats a good point. I'm wondering what alternatives we have, its sorta unfortunate that we used a string here, I wonder if there is some way we can find the pointer to this string at build time or something.

I think this fixes another bug here which is basically if you call one of the other syscalls with a buffer larger than 1024 it will break in an unexpected way, maybe even corrupt memory like we saw with this one.

Reply:
actually maybe i have an idea, we could pass some other flag through to let the host know we're doing this chunking and then check in the host that we're in sys_read.

Reply:
okay, interestingly the only syscall where this is a real problem I think is getenv